aiida/backends/tests/orm/data/test_data.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for the `Data` base class."""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import os
+import numpy
+
+from aiida import orm
+from aiida.backends.testbase import AiidaTestCase
+
+
+class TestData(AiidaTestCase):
+    """Test for the `Data` base class."""
+
+    @staticmethod
+    def generate_class_instance(data_class):
+        """Generate a dummy `Data` instance for the given sub class."""
+        dirpath_fixtures = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, 'fixtures'))
+        if data_class is orm.CifData:
+            instance = data_class(file=os.path.join(dirpath_fixtures, 'data', 'Si.cif'))
+            return instance
+
+        if data_class is orm.StructureData:
+            instance = orm.CifData(file=os.path.join(dirpath_fixtures, 'data', 'Si.cif')).get_structure()
+            return instance
+
+        if data_class is orm.BandsData:
+            kpoints = orm.KpointsData()
+            kpoints.set_cell([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+            kpoints.set_kpoints([[0., 0., 0.], [0.1, 0.1, 0.1]])
+            instance = data_class()
+            instance.set_kpointsdata(kpoints)
+            instance.set_bands([[1.0, 2.0], [3.0, 4.0]])
+            return instance
+
+        if data_class is orm.TrajectoryData:
+            instance = data_class()
+            stepids = numpy.array([60])
+            times = stepids * 0.01
+            cells = numpy.array([[[3., 0., 0.], [0., 3., 0.], [0., 0., 3.]]])
+            positions = numpy.array([[[0., 0., 0.]]])
+            instance.set_trajectory(stepids=stepids, cells=cells, symbols=['H'], positions=positions, times=times)
+            return instance
+
+        raise RuntimeError('no instance generator implemented for class `{}`'.format(data_class))
+
+    def test_data_exporters(self):
+        """Test the export functions for various data classes that should return content as bytes.
+
+        Note that the list of `Data` sub classes over which we loop is created manually here. If there are `Data`
+        classes that have export functionality that are not listed yet, they will not be tested. So they should be
+        added to the list manually.
+        """
+        for data_class in (orm.CifData, orm.StructureData, orm.BandsData, orm.TrajectoryData):
+
+            instance = self.generate_class_instance(data_class)
```

Comment:
yes, that seems like a good idea, since we can only test internal entry points anyway. The tests are broken due to #3412 again, but I think I have found the source, so I will make a separate PR for that